### PR TITLE
feat: instrument Phase 2a metrics on ingest

### LINKED ISSUE #17: Phase 2a: instrument minimum viable product metrics
## Goal

Implement the minimum viable Phase 2a metrics instrumentation in the OSS repo.

## Meta References

- Execution issue: https://github.com/tborys/driftshield-meta/issues/33
- Phase parent: https://github.com/tborys/driftshield-meta/issues/15
- Matching engine: https://github.com/tborys/driftshield-meta/issues/28
- Recurrence integration: https://github.com/tborys/driftshield-meta/issues/29
- Telemetry issue: https://github.com/tborys/driftshield-meta/issues/30
- Metrics semantics: https://github.com/tborys/driftshield-meta/issues/32

## Acceptance Criteria

- [x] The agreed minimum metrics are emitted or recorded end to end for the relevant Phase 2a flows
- [x] The instrumentation uses the agreed metric semantics rather than ad hoc naming
- [x] The resulting events or counters are documented clearly enough for later evidence review
- [x] The instrumentation path stays small enough to ship with the minimum slice

## Verification

- [x] A sample failure flow produces the expected metrics output
- [x] The observed output matches the definitions agreed in `driftshield-meta#32`

## Out Of Scope

- Building dashboards, BI pipelines, or large reporting systems
- Measuring every possible product interaction

## Agent Kickoff

Instrument only what the team has agreed matters for the Phase 2a evidence loop. Keep the path testable and obvious rather than comprehensive.

## Completion

Merged in PR #31: https://github.com/tborys/driftshield/pull/31

### PR DESCRIPTION
**Pull request by tborys**
## Summary
- instrument the OSS ingest flow to emit Phase 2a `analysis_result` telemetry events for newly analysed runs when telemetry opt-in is enabled
- derive the run-level telemetry fields from the analysed session result using the agreed minimal metric semantics
- document the real emission path and add regression tests for unclassified, matched, and deduped ingest behaviour

## Testing
- `PYTHONPATH=src pytest tests/api/test_ingest.py tests/cli/test_telemetry.py tests/cli/test_ingest.py tests/cli/test_connectors.py -q`
- `./scripts/check-public-scope.sh`

Closes #17

### CODE REVIEW

**Comment by tborys** [comment]
@codex review

**Comment by tborys** [comment]
Addressed the valid Codex finding.

Fixed now:
- move `analysis_result` telemetry emission out of `TranscriptIngestService` and into the `/api/ingest` route after a successful `db.commit()`
- keep `TranscriptIngestService` returning the analysed result so the route can emit metrics only for committed new ingests
- preserve connector watcher behaviour by updating it to the new ingest return shape without adding telemetry emission there
- add a regression test proving the duplicate-commit fallback does not emit a phantom metric event

Verification:
- `PYTHONPATH=src pytest tests/api/test_ingest.py tests/cli/test_telemetry.py tests/cli/test_ingest.py tests/cli/test_connectors.py -q`
- `./scripts/check-public-scope.sh`

**Comment by tborys** [comment]
@codex review

**Comment by tborys** [comment]
Addressed the latest valid Codex finding.

Fixed now:
- make post-commit telemetry emission best-effort so telemetry failures do not turn a successfully committed ingest into HTTP 422
- add a regression test proving ingest still returns success when telemetry emit raises after commit

Verification:
- `PYTHONPATH=src pytest tests/api/test_ingest.py tests/cli/test_telemetry.py tests/cli/test_ingest.py tests/cli/test_connectors.py -q`
- `./scripts/check-public-scope.sh`

**Comment by tborys** [comment]
@codex review

File: driftshield/src/driftshield/db/ingest_service.py [hunk 1]
```diff
@@ -67,14 +68,46 @@ def ingest_bytes(
...
+                domain_session,
+                result,
+                provenance,
+                existing_session_id=existing_session_id,
+            )
+
+        if not outcome.deduplicated:
+            metrics = _metrics_payload_from_analysis_result(result)
+            TelemetryService().record_analysis_event(
+                outcome_status=metrics["outcome_status"],
```
[the remark this answers is not in the document]
Comment (tborys) [reply driftshield/src/driftshield/db/ingest_service.py]: This is already addressed on `main`. Telemetry emission no longer happens inside `TranscriptIngestService.ingest_bytes`; the ingest route emits the event only after `db.commit()` succeeds, which avoids the phantom-metric path described here. Resolving as fixed.

File: driftshield/src/driftshield/api/routes/ingest.py [hunk 2]
```diff
@@ -38,12 +39,21 @@ def ingest_transcript(
...
-        outcome = ingest_service.ingest_bytes(
+        outcome, analysis_result = ingest_service.ingest_bytes(
             raw_bytes=raw_bytes,
             parser_name=normalised,
             source_path=file.filename,
         )
         db.commit()
+        if not outcome.deduplicated and analysis_result is not None:
+            metrics = metrics_payload_from_analysis_result(analysis_result)
+            TelemetryService().record_analysis_event(
```
[the remark this answers is not in the document]
Comment (tborys) [reply driftshield/src/driftshield/api/routes/ingest.py]: This is already addressed on `main`. The post-commit telemetry emission is now wrapped in its own best-effort `try/except`, so telemetry failures no longer change ingest success semantics after a successful commit. Resolving as fixed.
